enclave_manager/Dockerfile

```diff
@@ -137,61 +121,114 @@ RUN OPENSSL_VER=1.1.1d \
-             >> /etc/apt/apt.conf.d/00proxy; \
-        echo "http_proxy=$http_proxy" >> /etc/wgetrc; \
-        echo "http_proxy=$http_proxy" >> /etc/environment; \
-        echo "HTTP_PROXY=$(echo $http_proxy | sed 's,[a-zA-Z]*://,,')" \
-                  >> /etc/environment; \
-    fi \
- && if [ ! -z "$ftp_proxy"  ];  then \
-        echo 'Acquire::ftp::Proxy "'$ftp_proxy'";' \
-             >> /etc/apt/apt.conf.d/00proxy; \
-        echo "ftp_proxy=$ftp_proxy" >> /etc/wgetrc; \
-        echo "ftp_proxy=$ftp_proxy" >> /etc/environment; \
-    fi \
- && if [ ! -z "$https_proxy" ]; then \
-        echo 'Acquire::https::Proxy "'$https_proxy'";' \
-              >> /etc/apt/apt.conf.d/00proxy; \
-        echo "https_proxy=$https_proxy" >> /etc/wgetrc; \
-        echo "https_proxy=$https_proxy" >> /etc/environment; \
-        echo "HTTPS_PROXY=$(echo $https_proxy | sed 's,[a-zA-Z]*://,,')" \
-             >> /etc/environment; \
-    fi
-
 ENV http_proxy=$http_proxy
 ENV https_proxy=$https_proxy
-ENV TCF_HOME=/project/TrustedComputeFramework
+ENV TCF_HOME=/project/avalon
 ENV SGX_SSL=/opt/intel/sgxssl
 ENV SGX_SDK=/opt/intel/sgxsdk
 ENV PATH=$PATH:$SGX_SDK/bin:$SGX_SDK/bin/x64
 ENV PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$SGX_SDK/pkgconfig
 ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$SGX_SDK/sdk_libs
-# Set display, replace value with [IP address of host]:0
-ENV DISPLAY=:0
+ENV TCF_ENCLAVE_CODE_SIGN_PEM="$TCF_HOME/enclave.pem"
+
+RUN openssl genrsa -3 -out $TCF_HOME/enclave.pem 3072
+
+WORKDIR /project/avalon/common/sgx_workload
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/examples/apps
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/common/cpp
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/tc/sgx/trusted_worker_manager/common
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
+
+WORKDIR /project/avalon/tc/sgx/trusted_worker_manager/enclave
+
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
 
-WORKDIR /project/TrustedComputeFramework/tools/build
+WORKDIR /project/avalon/tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge
 
-RUN make lint && echo "LINT SUCCESS"
+RUN mkdir -p build \
+  && cd build \
+  && cmake .. \
+  && make
 
-RUN if [ "$MAKECLEAN" != 0 ]; then echo "Clean build" && make clean; fi \
- && make \
- && echo "BUILD SUCCESS"
+WORKDIR /project/avalon/common/python
+
+RUN make
+
+#ENV PATH=$PATH:/project/avalon/examples/enclave_manager
+
+WORKDIR /project/avalon/sdk
+
+RUN python3 setup.py bdist_wheel
+
+WORKDIR /project/avalon/enclave_manager
+
+RUN echo "Building avalon enclave manager" \
+  && make
+
+# Build Final image and install dependent modules
+FROM base_image as final_image
+
+ENV http_proxy=$http_proxy
+ENV https_proxy=$https_proxy
+ENV TCF_HOME=/project/avalon
+
+WORKDIR /project/avalon/
+
+COPY --from=build_image /usr/local/ /usr/local
```

Comment:
We can remove above line since wheel package is installed in final image

Reply:
This command is to copy the python modules that are downloaded during building stage. these modules are the run-time dependency for Avalon python modules. Hence copying them to final image.